packages/bridge-ui/src/components/form/SelectChain.svelte

```diff
@@ -2,15 +2,25 @@
   import ArrowRight from "../icons/ArrowRight.svelte";
   import { fromChain, toChain } from "../../store/chain";
   import { CHAIN_MAINNET, CHAIN_TKO } from "../../domain/chain";
+  import { ethereum } from "../../store/ethereum";
+  import { signer } from "../../store/signer";
+  import { switchEthereumChain } from "../../utils/switchEthereumChain";
+  import { ethers } from "ethers";
 
-  function toggleChains() {
-    fromChain.update((val) =>
-      val === CHAIN_MAINNET ? CHAIN_TKO : CHAIN_MAINNET
-    );
-    toChain.update((val) =>
-      val === CHAIN_MAINNET ? CHAIN_TKO : CHAIN_MAINNET
-    );
-  }
+  const toggleChains = async () => {
+    const chain = $fromChain === CHAIN_MAINNET ? CHAIN_TKO : CHAIN_MAINNET;
+    await switchEthereumChain($ethereum, chain);
+    const provider = new ethers.providers.Web3Provider(window.ethereum);
+    await provider.send("eth_requestAccounts", []);
+
+    fromChain.set(chain);
+    if (chain === CHAIN_MAINNET) {
+      toChain.set(CHAIN_TKO);
```

Comment:
nit: could reduce this to same ternary like `toChain.set(chain === CHAIN_MAINNET ? CHAIN_TKO : CHAIN_MAINNET);`